bookserver/session.py

```diff
@@ -41,3 +43,15 @@ async def load_user(user_id: str):
     """
     rslogger.debug(f"Going to fetch {user_id}")
     return await fetch_user(user_id)
+
+
+async def is_instructor(request: Request) -> bool:
+    user = request.state.user
+    if user is None:
+        raise HTTPException(401)
+    elif user.course_name in [
+        x.course_name for x in await fetch_instructor_courses(user.id)
```

Comment:
Wouldn't this be better as a part of the query, instead of getting all courses then searching the resulting list in Python?

Reply:
I knew you were going to say that. 😄 

My thinking here is that there are cases where we want all of the courses for an instructor, but it is a pretty small number for everyone but me and maybe one or two others.  So we are making and searching a list of less than six things almost all the time.

This makes for a more generally useful function.

Reply:
I guess the other alternative would be to have course_id as a secondary optional parameter for the function in crud. If provided we return zero or one objects and if its not provided we return all.  

I like the idea of a bunch of fairly generic calls in crud that do the CRUD things.  And I like the idea of isolating the database queries rather than spreading them everywhere.  But we'll have to see how it goes as we feel our way through the first bunch of API calls like this.